src/dag/write.rs

```diff
@@ -1,55 +1,195 @@
-use super::chunk::Chunk;
 use super::key::Key;
+use super::{chunk::Chunk, meta_generated::meta};
 use super::{read, Result};
 use crate::kv;
+use async_recursion::async_recursion;
+use async_std::sync::RwLock;
+use futures::future::try_join_all;
+use futures::try_join;
+use std::collections::HashSet;
+
+#[derive(Debug, Default)]
+struct HeadChange(Option<String>, Option<String>);
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+enum RefCount {
+    Inc = 1,
+    Dec = -1,
+}
 
 pub struct Write<'a> {
     kvw: Box<dyn kv::Write + 'a>,
+    changed_heads: RwLock<Vec<HeadChange>>,
```

Comment:
todo: is this changed, or just touched?

Reply:
Anything that was `put_chunked`. We do not check for no op puts

Reply:
Crap, the todo was a todo to myself to figure this out before sending the review. Nevermind this comment.

Reply:
I used to detect when the set_head was a no op, but it is very unlikely and the system works fine even if that was the case.